### PR TITLE
[SITES-13927] Franklin Forms Trial

### PR DESCRIPTION
**Pull request by swetabar**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #<gh-issue-id>
https://jira.corp.adobe.com/browse/SITES-14467
https://jira.corp.adobe.com/browse/SITES-13927
https://jira.corp.adobe.com/browse/SITES-14466

Test URLs:
- Before: https://main--aem-trials--adobe.hlx.page/drafts/swbarman/
- After: https://sites-13927-franklin-forms-trial--aem-trials--adobe.hlx.page/drafts/swbarman/

### CODE REVIEW

**Comment by blefebvre** [comment]
Can the gray bar be removed so it's just a plain white page? 
image

**Comment by blefebvre** [comment]
Can you please post the screenshots to slack for feedback from design and PM?

File: blocks/form/form.css [hunk 1]
```diff
@@ -0,0 +1,125 @@
+main .form {
+    background-color: white;
```
Comment (blefebvre) [inline blocks/form/form.css:2]: Indentation looks off in this file.

**Comment by blefebvre** [comment]
Form seems stuck in its "invalid" state, even when I change to a valid email and re-submit: 
image

File: blocks/form/checkbox.svg
Comment (blefebvre) [inline blocks/form/checkbox.svg:1]: What is the checkbox used for?

**Review by blefebvre — CHANGES_REQUESTED** [review]

**Comment by swetabar** [comment]
> Form seems stuck in its "invalid" state, even when I change to a valid email and re-submit: image

Something weird happened. Might have been after the refactoring. I'll take a peek.

File: blocks/form/checkbox.svg
Comment (swetabar) [reply blocks/form/checkbox.svg:1]: I'll remove it. It's from the form I picked up from Franklin.

File: blocks/form/form.css (hunk 1 above)
Comment (swetabar) [reply blocks/form/form.css]: Modified.

File: blocks/form/checkbox.svg
Comment (swetabar) [reply blocks/form/checkbox.svg:1]: Removed.

**Comment by swetabar** [comment]
> Can the gray bar be removed so it's just a plain white page? image

Fixed.

File: blocks/form/form.css [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+  height: 30px;
+  width: 30px;
+  position: relative;
+  border: unset;
+  padding: unset;
+  margin-right: 5px;
+  margin-left: 0;
+}
+
+main .form .form-checkbox-wrapper input[type='checkbox']::after {
```
Comment (blefebvre) [inline blocks/form/form.css:93]: When do we show a checkbox?

File: blocks/form/form.js [hunk 3]
```diff
@@ -0,0 +1,217 @@
+function createSelect(fd) {
```
Comment (blefebvre) [inline blocks/form/form.js:1]: Please add a comment to this file indicating its purpose & use.

File: blocks/form/form.js [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+}
+
+function checkValidity(form) {
+  const payload = constructPayload(form);
+  const { email } = payload;
+  const regex = /^[^@]+@(?!(yahoo|hotmail|gmail|icloud|outlook))[^@]+\.[a-z]{2,}$/;
+  return String(email).match(regex);
+}
+
+function createButton(fd) {
```
Comment (blefebvre) [inline blocks/form/form.js:57]: What is `fd`? Can these variable names be changed to be human readable?

Can a comment please be added indicating what these functions are used for?

File: blocks/form/form.js [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+  const button = document.createElement('button');
+  button.textContent = fd.Label;
+  button.classList.add('button');
+  button.classList.add('primary');
+  if (fd.Type === 'submit') {
+    button.addEventListener('click', async (event) => {
+      const form = button.closest('form');
+      if (fd.Placeholder) form.dataset.action = fd.Placeholder;
+      if (form.checkValidity() && checkValidity(form) !== null) {
+        event.preventDefault();
```
Comment (blefebvre) [inline blocks/form/form.js:67]: Looks like this is duplicated in both sides of the `if`.

File: blocks/form/form.js [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+      case 'text-area':
+        fieldWrapper.append(createLabel(fd));
+        fieldWrapper.append(createTextArea(fd));
+        break;
+      case 'submit':
+        fieldWrapper.append(createButton(fd));
+        break;
+      default:
+        // For now, we are not creating any label for the email field
+        //   fieldWrapper.append(createLabel(fd));
```
Comment (blefebvre) [inline blocks/form/form.js:191]: Remove commented code.

File: blocks/form/form.js [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+        fieldWrapper.append(createInput(fd));
+    }
+
+    if (fd.Rules) {
+      try {
+        rules.push({ fieldId, rule: JSON.parse(fd.Rules) });
+      } catch (e) {
+        // eslint-disable-next-line no-console
+        console.warn(`Invalid Rule ${fd.Rules}: ${e}`);
```
Comment (blefebvre) [inline blocks/form/form.js:200]: What are the `Rules`?

File: blocks/form/form.css (hunk 2 above)
Comment (swetabar) [reply blocks/form/form.css]: Removed.

File: blocks/form/form.js (hunk 3 above)
Comment (swetabar) [reply blocks/form/form.js]: Removed this method.

File: blocks/form/form.js (hunk 4 above)
Comment (swetabar) [reply blocks/form/form.js]: Modified + added comments.

File: blocks/form/form.js (hunk 5 above)
Comment (swetabar) [reply blocks/form/form.js:56]: Done.

File: blocks/form/form.js (hunk 6 above)
Comment (swetabar) [reply blocks/form/form.js]: Done.

File: blocks/form/form.js (hunk 7 above)
Comment (swetabar) [reply blocks/form/form.js]: We're not using Rules for this form. Removed it.

File: blocks/form/form.js (hunk 5 above)
Comment (blefebvre) [reply blocks/form/form.js:56]: I still see `event.preventDefault();` duplicated in the `if` and the `else`. It can be moved above to avoid duplication.

Comment (swetabar) [reply blocks/form/form.js]: Done.

**Pull request merged**